pkgs/development/python-modules/semaphore/default.nix

```
@@ -0,0 +1,37 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, python-dateutil
+, attrs
+, anyio
+}:
+
+buildPythonPackage rec {
+  pname = "semaphore";
+  version = "0.16.0";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "semaphore-bot";
+    hash = "sha256-EOUvzW4a8CgyQSxb2fXnIWfOYs5Xe0v794vDIruSHmI=";
+  };
+
+  postPatch = ''
+    substituteInPlace setup.py --replace "anyio>=3.5.0,<=3.6.2" "anyio"
+  '';
+
+  propagatedBuildInputs = [
+    anyio
+    attrs
+    python-dateutil
+  ];
+
+  pythonImportsCheck = [ "semaphore" ];
+
+  meta = with lib; {
+    description = "Simple rule-based bot library for Signal Private Messenger";
+    homepage = "https://github.com/lwesterhof/semaphore";
+    license = with licenses; [ agpl3 ];
```

```suggestion
    license = with licenses; [ agpl3Plus ];
```
https://github.com/lwesterhof/semaphore/blob/b1914a60e3cbfbf682b1fa61e91b51f144e83192/semaphore/__init__.py#L6-L9